File: 2024-2025/domaci_kolo/templates/A.cs
// TODO: Uprav tuto funkci
int MinimalniCas(int pocetKarticek, int strihackaCas, int nuzkyCas, int nuzkyPocet)
{
    return 5;  // Vrať celkový potřebný čás
}

// Načítání vstupu

void PrectiVstup(TextReader vstup, TextWriter vystup)
{
    int pocetProblemu = int.Parse(vstup.ReadLine()!);

    for (int i = 0; i < pocetProblemu; i++)
    {
        string[] problem = vstup.ReadLine()!.Split(' ');
        int pocetKarticek = int.Parse(problem[0]);
        int strihackaCas = int.Parse(problem[1]);
        int nuzkyCas = int.Parse(problem[2]);
        int nuzkyPocet = int.Parse(problem[3]);

        vystup.WriteLine(MinimalniCas(pocetKarticek, strihackaCas, nuzkyCas, nuzkyPocet));
    }
}

(string, string)[] nazvySouboru = {
    ("A-lehky.txt", "A-lehky-vystup.txt"),
    ("A-tezky.txt", "A-tezky-vystup.txt"),
};

bool souborNalezen = false;

foreach (var (nazevVstupu, nazevVystupu) in nazvySouboru)
{
    try
    {
        using var vstup = new StreamReader(File.OpenRead(nazevVstupu));
        using var vystup = new StreamWriter(File.OpenWrite(nazevVystupu));

        PrectiVstup(vstup, vystup);

        souborNalezen = true;
    }
    catch (FileNotFoundException) { }
}

if (!souborNalezen)
{
    PrectiVstup(Console.In, Console.Out);
}

File: 2024-2025/domaci_kolo/templates/A.py
#!/usr/bin/env python3
import sys


def minimalni_cas(pocet_karticek, strihacka_cas, nuzky_cas, nuzky_pocet):
    """TODO: Uprav tuto funkci"""

    return 5  # Vrať celkový potřebný čás


# Načítání vstupu


def precti_vstup(input, output):
    pocet_problemu = int(input.readline())

    for _ in range(pocet_problemu):
        pocet_karticek, strihacka_cas, nuzky_cas, nuzky_pocet = map(int, input.readline().split())

        reseni = minimalni_cas(pocet_karticek, strihacka_cas, nuzky_cas, nuzky_pocet)

        output.write(f"{reseni}\n")


NAZVY_SOUBORU = [
    ("A-lehky.txt", "A-lehky-vystup.txt"),
    ("A-tezky.txt", "A-tezky-vystup.txt"),
]

soubor_nalezen = False

for nazev_vstupu, nazev_vystupu in NAZVY_SOUBORU:
    try:
        with open(nazev_vstupu, "r", 1, "utf-8") as vstup:
            with open(nazev_vystupu, "w", 1, "utf-8") as vystup:
                precti_vstup(vstup, vystup)

        soubor_nalezen = True

    except FileNotFoundError:
        pass

if not soubor_nalezen:
    precti_vstup(sys.stdin, sys.stdout)
